src/bin/typos-cli/report.rs

```diff
@@ -284,3 +298,68 @@ impl Report for PrintJson {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::report::calculate_visible_column_width;
+
+    #[test]
+    fn test_calculate_visible_column_width() {
```

Comment:
nit: I'd personally do a test per `*_chars` array, with the array name being moved to the test name.  This makes it easier to see more test results rather than having to work on a failure at a time